membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -0,0 +1,144 @@
+package services
+import com.gu.memsub.Subscription.AccountId
+import com.gu.memsub.subsv2.Subscription
+import com.gu.memsub.subsv2.SubscriptionPlan.AnyPlan
+import com.gu.memsub.subsv2.reads.SubPlanReads
+import com.gu.memsub.subsv2.reads.SubPlanReads.anyPlanReads
+import com.gu.zuora.ZuoraRestService.QueryResponse
+import loghandling.LoggingField.LogField
+import loghandling.{LoggingWithLogstashFields, ZuoraRequestCounter}
+import models.Attributes
+import org.joda.time.LocalDate
+import play.api.libs.concurrent.Execution.Implicits._
+
+import scala.concurrent.Future
+import scalaz.std.list._
+import scalaz.std.scalaFuture._
+import scalaz.syntax.traverse._
+import scalaz.{-\/, Disjunction, EitherT, \/, \/-, _}
+
+
+object AttributesFromZuora extends LoggingWithLogstashFields {
+
+  def getAttributes(identityId: String,
+                    identityIdToAccountIds: String => Future[String \/ QueryResponse],
+                    subscriptionsForAccountId: AccountId => SubPlanReads[AnyPlan] => Future[Disjunction[String, List[Subscription[AnyPlan]]]],
+                    dynamoAttributeGetter: String => Future[Option[Attributes]]): Future[Option[Attributes]] = {
```

Comment:
great! this feels very self contained we could even reuse this in a new project without a touchpoint backend!
there are a fair few parameters but hopefully we won't need the dynamoAttributeGetter in the long run.